scalafix-rules/src/main/scala/scala/meta/internal/pc/ScalafixGlobal.scala

```diff
@@ -169,6 +169,33 @@ class ScalafixGlobal(
       .filterKeys(_ != NoSymbol)
       .toMap
 
+  // currently wrapping in backticks supported only for traits and case classes
+  private def backtickify(
+    tpe: Type,
+    loop: (Type, Option[ShortName]) => Type,
+    withPrefix: Boolean
+  ): Type = {
+    val supportedTypes: Symbol => Boolean =
+      s => s.isCaseClass || s.isTrait
```

Comment:
I am still curious to know what's the limitation with other simple types like a regular class?

Reply:
Changing `isCaseClass` -> `isClass` leads to failed test on `(=> Any) => Any` case. So, it brings redundant backticks. And this could mean there are other cases when it also would lead to redundant backticks (which are missing in unit tests).

Reply:
Though, maybe it's not so bad to add backticks for the `(=> Any) => Any`...  I'm just not sure wouldn't bring other uncontrolled changes in current behavior.

Reply:
> So, it brings redundant backticks

I removed the `supportedTypes` filter to have a look at it, and the problem actually seems to be that we can't rely on `decodedName` for by-names.
```
###########> Diff       <###########

=======
=> Diff
=======
--- obtained
+++ expected
@@ -133,3 +133,3 @@
∙
-    val byNameType: `<byname>`[Any] => Any = ??? : ((=> Any) => Any)
+    val byNameType: (=> Any) => Any = ??? : ((=> Any) => Any)
     // FIXME: https://github.com/twitter/rsc/issues/144
         
[info] - test/explicitResultTypes/RscCompat.scala *** FAILED *** (273 milliseconds)
[info]   see above (AbstractSemanticRuleSuite.scala:91)
```

As a result, I took the liberty to add b2c88064a00cafe496cd84bf40187ff84caaf3c0 to your branch to unlock support for classes as well. Let me know what you think!

Reply:
Whoa! That's tremendous. Thank you so much.